jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/spi/AbsolutePathURLProvider.java

```diff
@@ -55,10 +57,28 @@ public byte[] getContent(final String url, final String urlPrefix) {
             String line;
             while (null != (line = reader.readLine())) {
                 if (!line.startsWith("#")) {
+                    line = replaceVariables(line);
                     builder.append(line).append('\n');
                 }
             }
             return builder.toString().getBytes(StandardCharsets.UTF_8);
         }
     }
+    
+    private String replaceVariables(final String line) {
+        Pattern variablePattern = Pattern.compile("\\$\\{(.+?)\\}");
```

Comment:
- The regular expression `${(.+?)}` appears to conflict with the syntax involved in the Groovy implementation of Row Value Expressions. Maybe replace it with `#{(.+?)}`? Refer to https://shardingsphere.apache.org/document/current/en/user-manual/common-config/builtin-algorithm/expr/ .
- Or other grammar recognition?
- I added separate documentation in #29730, it seems good to add two separate implementation for `#{(.+?)}`, a non-standard way of identifying environment variables. What do you think?

Reply:
- I should know where the conflict is going to be, but using `#{(.+?)}` would be considered a comment, so my current approach may not work well.